src/lib/configure_dynatrace.py

```diff
@@ -0,0 +1,123 @@
+            self.logging_context.log(f'Unable to get existing dashboards config. Error details: {e}')
+            return {}
+
+
+    async def get_available_dashboards(self) -> dict:
+        try:
+            selected_services = None
+            if "GCP_SERVICES" in os.environ:
+                selected_services_string = os.environ.get("GCP_SERVICES", "")
+                selected_services = selected_services_string.split(",") if selected_services_string else []
+                working_directory = os.path.dirname(os.path.realpath(__file__))
+                config_directory = os.path.join(working_directory, "../config")
+                config_files = [
+                    file for file
+                    in listdir(config_directory)
+                    if isfile(os.path.join(config_directory, file)) and is_yaml_file(file)
+                ]
+                dashboards = []
+                for file in config_files:
+                    config_file_path = os.path.join(config_directory, file)
+                    try:
+                        with open(config_file_path, encoding="utf-8") as config_file:
+                            config_yaml = yaml.safe_load(config_file)
+                            for dashboard in config_yaml.get("dashboards", []):                        
+                                dashboard_file_path = os.path.join(working_directory, '../', dashboard.get("dashboard",{}))                                                                
+                                with open(dashboard_file_path, encoding="utf-8") as dashboard_file:
+                                    dashboard_json = json.load(dashboard_file) 
+                                    dashboard_name = dashboard_json.get("dashboardMetadata",{}).get("name",{})
+                                    dashboards.append({"path": dashboard_file_path,"name": dashboard_name})                            
+                    except Exception as error:
+                         self.logging_context.log(f"Failed to load configuration file: '{config_file_path}'. Error details: {error}")
+                         continue                
+                return dashboards            
+        except Exception as e:
+            self.logging_context.log(f'Unable to get available dashboards. Error details: {e}')
+            return {}
+            
+
+    def __init__(self, session: ClientSession, logging_context: LoggingContext):
+        self.session = session
+        self.logging_context = logging_context
+            
+    async def _init_(self):        
+        dynatrace_url = await fetch_dynatrace_url(self.session, "", "")
+        dynatrace_access_key = await fetch_dynatrace_api_key(self.session,"", "")
+        scopes = await  get_dynatrace_token_metadata(self.session, self.logging_context, dynatrace_url, dynatrace_access_key)
+        has_write_config_permission=any(s in scopes.get('scopes') for s in ["WriteConfig", "ReadConfig"])        

```

Comment:
Please use `scopes.get('scopes', [])` to avoid KeyError if `get_dynatrace_token_metadata` returns empty dict on request exception